### ISSUE TITLE
OpSimPySM does not convert coordinates properly for Q/U components

### ISSUE THREAD

**Issue by ziotom78** [hpc4cmb/toast#242]
As it is written now, `OpSimPySM.exec` converts PySM maps from Galactic to XXX coordinates using a rotator and a call to `healpy.get_interp_val` (line [193](https://github.com/hpc4cmb/toast/blob/master/src/python/todmap/pysm_operator.py#L195)). However, it does not recalculate the value of Q and U in the new coordinate system, using a call to `healpy.rotator.Rotator.angle_ref`.

I am not fully sure of how this should be implemented. I believe that the calculation should proceed along the following lines: if `q_map` and `u_map` are the Q and U maps, then
```python
# This is a *direct* rotator, from Galactic to whatever
rot = healpy.rotator.Rotator(coord=["G", self.coord])

# Add the angle calculated by angle_ref to the argument of the L parameter (complex number)
L_map = (q_map + u_map * 1j) * numpy.exp(1j * rot.angle_ref(theta_gal, phi_gal))

# Overwrite the Q and U maps with the correct values
q_map = numpy.real(L_map)
u_map = numpy.imag(L_map
```
where `L_map` is a map of the complex-valued L parameter reported in the [Wikipedia page about Stokes parameters](https://en.wikipedia.org/wiki/Stokes_parameters#Properties).

This should work for some definition of `theta_gal` and `phi_gal`. I'm not sure of this however, as I've just discovered the existence of `healpy.angle_ref`. Thoughts?

**Comment by zonca** [comment]
thanks @ziotom78, do you have a test case at hand that we can use to check if the implementation is correct?
Otherwise I'll build one from one of the `healpy` test cases.

**Comment by zonca** [comment]
also we can check the IDL implementation in HEALPix https://github.com/healpy/healpixmirror/blob/master/src/idl/misc/rotate_coord.pro

**Comment by ziotom78** [comment]
Sorry, I have no test cases at hand. I realized this last Friday while discussing with a student of mine at the blackboard.

**Comment by zonca** [comment]
thanks, it was missing a factor of 2:

    L_map = (q_map + u_map * 1j) * numpy.exp(1j * 2 * rot.angle_ref(theta_gal, phi_gal))

I ran IDL and compared this to `rotate_coord.pro`, it gets the same results within 1e-4.

I wrote the implementation for `healpy` in https://github.com/healpy/healpy/pull/447

**Comment by zonca** [comment]
implementing a fix in #246

as expected, in the test, the total Polarization amplitude is unchanged, but Q and U are rotated.

![q_diff](https://user-images.githubusercontent.com/383090/41306011-0c3312b8-6e75-11e8-8baa-628fecf4bc99.png)
![total_p_diff](https://user-images.githubusercontent.com/383090/41306012-0c66cf36-6e75-11e8-97df-8fb63e756fa5.png)
